.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,58 @@
+name: pull-request
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+    - source/**
+    - clik/**
+    - modules/**
+    - examples/**
+    - cmake/**
+    - hal/**
+    - .github/actions/do_build_ock/**
+    - .github/actions/setup_ubuntu_build/**
+    - .github/workflows/run_pr_tests.yml
+    - .github/workflows/pull-request.yml
+    - CMakeLists.txt
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+jobs:
+
+############### JOB   mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+
+  mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 60
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.1.0
+      with:
+        path: code
```

Comment:
still not really sure why we need this, since we don't have it in our other jobs.

Reply:
As Gitlab uses `build.py` the model used there doesn't migrate unchanged. Its not shown in this job, but some Gitlab jobs also create build dirs outwith the workspace with `build.py` jumping in and out of the workspace and using command line settings to point it back to "source" to perform parts of the runs. 
I tried to simplify that so Github jobs do all their work in the workspace with "top-level" dirs in there for various purposes to stop things becoming cluttered: i.e. code, build, install, llvm, any "extra dirs" the build needs, and probably other things I haven't met yet. It also helps to work out how all the pieces fit together during job development.
However ... much of it comes down to personal preference of course so I have no objections if you want to suggest an alternative scheme.

Reply:
I don't have strong opinions, it just feels like a departure from what we have and calling it code - just means oneapi-construction-kit will be checked out as code - is that really clearer?

Reply:
There's no build.py now so it will be different. I used "code" as that's the name Github uses to refer to repo content but we can rename that if required. I think overall its clearer as the structure is more consistent between jobs and everything is under the workspace (although admittedly I've not looked at all jobs as yet).

Reply:
Closing in favour of a later conversation.